File: tree.java
import java.util.stream.IntStream;

public class tree{

	public static void magicalTree(Integer levels){
		if(levels>0){
			IntStream.range(1, levels+1 ).forEach(level->{
				IntStream.range(1, levels-level+1).forEach(v->{
					System.out.print(" ");
				});
				IntStream.range(0, 2*(level-1)+1).forEach(v->{
					System.out.print("*");
				});
				System.out.println("");
			});
		}
	}

	public static void main(String[]args){
		Integer levels;
		if(args.length!=1){
			System.out.println("You must provide one integer argument");
			System.exit(-1);
		}
		
		try{
			magicalTree(Integer.parseInt(args[0]));

		}
		catch(NumberFormatException e){
			System.out.println("Error parsing the argument, exiting");
			System.exit(-1);
		}	
	}
}

File: tree.py
#!/usr/bin/env python3

import argparse
import sys

class ownParser(argparse.ArgumentParser):
    def error(self, message):
        sys.stderr.write('error: %s\n\n' % message)
        self.print_help()
        sys.exit(1)

def magical_tree(height):
    for level in range(1,height+1):
        for spaces in range(1,height-level+1):
            print(" ", end = "")
        for balls in range(0, 2*(level-1)+1):
            print("*", end = "")
        print("")

if __name__ == '__main__':
    exit_code         = 0

    parser = ownParser(description='Build a magical tree')
    parser.add_argument('-l', '--levels', nargs='?', type=int, default=[], required=True)

    args=parser.parse_args()

    magical_tree(args.__dict__["levels"])
